### PR TITLE
Demo Docs: Update the name of recommendation cache feature flag

### PR DESCRIPTION
**Pull request by cthain**
This PR updates the demo docs to reference the correct name of the feature flag for the recommendation service cache failure. It's not a significant change but brings the docs in line with the flag name in the `demo.flagd.json` file.

### CODE REVIEW

**Review by julianocosta89 — APPROVED** [review]
Thank you @cthain!

**Comment by cthain** [comment]
No worries. Before this gets merged, do you any input on my comment about the feature flags when running in kube: https://github.com/open-telemetry/opentelemetry-demo/issues/1343#issuecomment-1996007882? I'm happy to augment this PR with k8s-specific info if you think that would be helpful.

It might also be helpful to harmonize on the flag names, since they are currently named differently in the json file (docker compose) vs the ffspostgres (k8s)

**Review by theletterf — APPROVED** [review]
Thanks!

File: content/en/docs/demo/feature-flags.md [hunk 1]
```diff
@@ -13,14 +13,14 @@ feature flag service that supports [OpenFeature](https://openfeature.dev). Flag
...
-| `paymentServiceUnreachable` | Checkout Service | Use a bad address when calling the PaymentService to make it seem like the PaymentService is unavailable. |
+| Feature Flag                        | Service(s)      | Description                                                                                              |
+| ----------------------------------- | --------------- | -------------------------------------------------------------------------------------------------------- |
+| `adServiceFailure`                  | Ad Service      | Generate an error for `GetAds` 1/10th of the time                                                        |
+| `cartServiceFailure`                | Cart Service    | Generate an error for `EmptyCart` 1/10th of the time                                                     |
+| `productCatalogFailure`             | Product Catalog | Generate an error for `GetProduct` requests with product id: `OLJCESPC7Z`                                |
+| `recommendationServiceCacheFailure` | Recommendation  | Create a memory leak due to an exponentially growing cache. 1.4x growth, 50% of requests trigger growth. |
```
Comment (cartermp) [inline content/en/docs/demo/feature-flags.md:21]:
```suggestion
| Feature Flag                        | Service(s)       | Description                                                                                              |
| ----------------------------------- | ---------------- | -------------------------------------------------------------------------------------------------------- |
| `adServiceFailure`                  | Ad Service       | Generate an error for `GetAds` 1/10th of the time                                                        |
| `cartServiceFailure`                | Cart Service     | Generate an error for `EmptyCart` 1/10th of the time                                                     |
| `productCatalogFailure`             | Product Catalog  | Generate an error for `GetProduct` requests with product id: `OLJCESPC7Z`                                |
| `recommendationServiceCacheFailure` | Recommendation   | Create a memory leak due to an exponentially growing cache. 1.4x growth, 50% of requests trigger growth. |
```

**Review by cartermp — APPROVED** [review]
final whitespace changes

**Pull request merged**